exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/NumericEquiDepthHistogram.java

```diff
@@ -69,27 +79,177 @@ public void setNumRowsPerBucket(long numRows) {
   }
 
   @Override
-  public Double estimatedSelectivity(RexNode filter) {
+  public Double estimatedSelectivity(final RexNode filter) {
     if (numRowsPerBucket >= 0) {
-      return 1.0;
-    } else {
-      return null;
+      // at a minimum, the histogram should have a start and end point of 1 bucket, so at least 2 entries
+      Preconditions.checkArgument(buckets.length >= 2,  "Histogram has invalid number of entries");
+      final int first = 0;
+      final int last = buckets.length - 1;
+
+      // number of buckets is 1 less than the total # entries in the buckets array since last
+      // entry is the end point of the last bucket
+      final int numBuckets = buckets.length - 1;
+      final long totalRows = numBuckets * numRowsPerBucket;
+      if (filter instanceof RexCall) {
+        // get the operator
+        SqlOperator op = ((RexCall) filter).getOperator();
+        if (op.getKind() == SqlKind.GREATER_THAN ||
+                op.getKind() == SqlKind.GREATER_THAN_OR_EQUAL) {
+          Double value = getLiteralValue(filter);
+          if (value != null) {
+
+            // *** Handle the boundary conditions first ***
+
+            // if value is less than or equal to the first bucket's start point then all rows qualify
+            int result = value.compareTo(buckets[first]);
+            if (result <= 0) {
+              return LARGE_SELECTIVITY;
+            }
+            // if value is greater than the end point of the last bucket, then none of the rows qualify
+            result = value.compareTo(buckets[last]);
+            if (result > 0) {
+              return SMALL_SELECTIVITY;
```

Comment:
For posterity, please add an explanation for SMALL_SELECTIVITY > 0 and the meaning of buckets[n], if not added elsewhere.

Reply:
Done.  I added a comment when we declare the class variable.